cmd/agent/subcommands/secret/command.go

```diff
@@ -25,39 +25,58 @@ import (
 // cliParams are the command-line arguments for this subcommand
 type cliParams struct {
 	*command.GlobalParams
+
+	// args are the positional command-line arguments
+	args []string
+
+	// cmd is the cobra Command, used to show help
+	cmd *cobra.Command
 }
 
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 	cliParams := &cliParams{
 		GlobalParams: globalParams,
 	}
-	secretInfoCommand := &cobra.Command{
-		Use:   "secret",
-		Short: "Print information about decrypted secrets in configuration.",
+
+	cmd := &cobra.Command{
+		Use:   "secret [refresh]",
+		Short: "Display information about or refresh secrets in configuration.",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return fxutil.OneShot(secret,
+			cliParams.cmd = cmd
+			cliParams.args = args
+			return fxutil.OneShot(secretCommand,
 				fx.Supply(cliParams),
 				fx.Supply(command.GetDefaultCoreBundleParams(cliParams.GlobalParams)),
 				core.Bundle(),
 			)
 		},
 	}
 
-	return []*cobra.Command{secretInfoCommand}
+	return []*cobra.Command{cmd}
```

Comment:
What do you think about using a [hierarchical command](https://github.com/DataDog/datadog-agent/blob/7.50.0/cmd/agent/subcommands/jmx/command.go#L201-L209). You could easily add parameters to `secret refresh` command and provides a message for `agent secret` and another one for `agent secret refresh`. It also removes the need for https://github.com/DataDog/datadog-agent/pull/22031/files#diff-b0775b8561ff974b97467944c975f5000f61de732f51008a77cd9bc7e923e93bR66-R77. What do you think?

Reply:
That's much better, thanks for the suggestion!